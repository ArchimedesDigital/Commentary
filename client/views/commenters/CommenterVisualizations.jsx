
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

CommenterVisualizations = React.createClass({

	propTypes: {
		commenter: React.PropTypes.object.isRequired,
	},

	childContextTypes: {
		muiTheme: React.PropTypes.object.isRequired,
	},

	getInitialState() {
		return {
			visibleWork: '',
		};
	},

	getChildContext() {
		return { muiTheme: getMuiTheme(baseTheme) };
	},

	toggleVisibleWork(workToToggle) {
		if (this.state.visibleWork === workToToggle) {
			this.setState({
				visibleWork: '',
			});
		} else {
			this.setState({
				visibleWork: workToToggle,
			});
		}
	},

	renderWorks() {
		return this.props.commenter.nCommentsWorks.map((work, i) =>
			<CommenterWorkVisualization
				key={i}
				toggleVisibleWork={this.toggleVisibleWork}
				work={work}
			/>
		);
	},

	render() {
		const commenter = this.props.commenter;
		let classes = 'commenter-visualizations';
		if (this.state.visibleWork.length) {
			classes += ` commenter-visualizations-${this.state.visibleWork}-visible`;
		}

		let workIliadLevel = 0;
		let workOdysseyLevel = 0;
		let workHymnsLevel = 0;

		if (commenter.nCommentsIliad > 0) {
			workIliadLevel = Math.floor((commenter.nCommentsIliad / commenter.nCommentsTotal) * 10);
		}

		if (commenter.nCommentsOdyssey > 0) {
			workOdysseyLevel = Math.floor((commenter.nCommentsOdyssey / commenter.nCommentsTotal) * 10);
		}

		if (commenter.nCommentsHymns > 0) {
			workHymnsLevel = Math.floor((commenter.nCommentsHymns / commenter.nCommentsTotal) * 10);
		}
		console.log(workIliadLevel);

		return (
			<div className={classes}>
				<div className="commenter-visualization-title">
					<h2>
						Comments
					</h2>
				</div>
				<div className="commenter-work-circles">
					{workIliadLevel ?
						<CommenterWorkCircle
							toggleVisibleWork={this.toggleVisibleWork}
							workTitle={"Iliad"}
							workSlug={"iliad"}
							workLevel={workIliadLevel}
						/>
					: '' }
					{workOdysseyLevel ?
						<CommenterWorkCircle
							toggleVisibleWork={this.toggleVisibleWork}
							workTitle={"Odyssey"}
							workSlug={"odyssey"}
							workLevel={workOdysseyLevel}
						/>
					: '' }
					{workHymnsLevel ?
						<CommenterWorkCircle
							toggleVisibleWork={this.toggleVisibleWork}
							workTitle={"Hymns"}
							workSlug={"homeric-hymns"}
							workLevel={workHymnsLevel}
						/>
					: '' }
				</div>
				<div className="work-visualizations">
					{this.renderWorks()}
				</div>
			</div>
		);
	},
});
