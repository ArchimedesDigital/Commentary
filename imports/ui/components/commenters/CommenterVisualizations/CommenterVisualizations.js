import React from 'react';

import getMuiTheme from 'material-ui/styles/getMuiTheme';

// lib:
import muiTheme from '/imports/lib/muiTheme';

// components
import CommenterWorkCircle from '/imports/ui/components/commenters/CommenterWorkCircle';
import CommenterWorkVisualization from '/imports/ui/components/commenters/CommenterWorkVisualization';


const CommenterVisualizations = React.createClass({

	propTypes: {
		commenter: React.PropTypes.object.isRequired,
		isTest: React.PropTypes.bool,
	},

	childContextTypes: {
		muiTheme: React.PropTypes.object.isRequired,
	},

	getInitialState() {
		return {
			visibleWork: '',
		};
	},

	getChildContext() {
		return { muiTheme: getMuiTheme(muiTheme) };
	},

	toggleVisibleWork(workToToggle) {
		if (this.state.visibleWork === workToToggle) {
			this.setState({
				visibleWork: '',
			});
		} else {
			this.setState({
				visibleWork: workToToggle,
			});
		}
	},

	renderWorks() {
		const { commenter, isTest } = this.props;

		if (commenter.nCommentsWorks) {
			return commenter.nCommentsWorks.map((work, i) =>
				<CommenterWorkVisualization
					key={i}
					toggleVisibleWork={this.toggleVisibleWork}
					work={work}
					commenterSlug={commenter.slug}
					isTest={isTest}
				/>
			);
		}
		return '';
	},

	render() {
		const { commenter } = this.props;

		let classes = 'commenter-visualizations';
		if (this.state.visibleWork.length) {
			classes += ` commenter-visualizations-${this.state.visibleWork}-visible`;
		}

		let workIliadLevel = 0;
		let workOdysseyLevel = 0;
		let workHymnsLevel = 0;

		if (commenter.nCommentsIliad > 0) {
			workIliadLevel = Math.floor((commenter.nCommentsIliad / commenter.nCommentsTotal) * 10);
		}

		if (commenter.nCommentsOdyssey > 0) {
			workOdysseyLevel = Math.floor((commenter.nCommentsOdyssey / commenter.nCommentsTotal) * 10);
		}

		if (commenter.nCommentsHymns > 0) {
			workHymnsLevel = Math.floor((commenter.nCommentsHymns / commenter.nCommentsTotal) * 10);
		}

		return (
			<div className={classes}>
				<div className="commenter-visualization-title">
					<h2>
						Comments
					</h2>
				</div>
				<div className="commenter-work-circles">
					{workIliadLevel ?
						<CommenterWorkCircle
							toggleVisibleWork={this.toggleVisibleWork}
							workTitle={'Iliad'}
							workSlug={'iliad'}
							workLevel={workIliadLevel}
							nComments={commenter.nCommentsIliad}
						/>
						: '' }
					{workOdysseyLevel ?
						<CommenterWorkCircle
							toggleVisibleWork={this.toggleVisibleWork}
							workTitle={'Odyssey'}
							workSlug={'odyssey'}
							workLevel={workOdysseyLevel}
							nComments={commenter.nCommentsOdyssey}
						/>
						: '' }
					{workHymnsLevel ?
						<CommenterWorkCircle
							toggleVisibleWork={this.toggleVisibleWork}
							workTitle={'Hymns'}
							workSlug={'homeric-hymns'}
							workLevel={workHymnsLevel}
							nComments={commenter.nCommentsHymns}
						/>
						: '' }
				</div>
				<div className="work-visualizations">
					{this.renderWorks()}
				</div>
			</div>
		);
	},
});

export default CommenterVisualizations;
