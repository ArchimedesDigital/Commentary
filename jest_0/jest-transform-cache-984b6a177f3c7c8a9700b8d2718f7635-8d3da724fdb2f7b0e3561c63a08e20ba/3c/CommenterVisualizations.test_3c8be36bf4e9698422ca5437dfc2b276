/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_reactTestRenderer = require('react-test-renderer'); /* istanbul ignore next */var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);


var /* istanbul ignore next */_CommenterVisualizations = require('./CommenterVisualizations'); /* istanbul ignore next */var _CommenterVisualizations2 = _interopRequireDefault(_CommenterVisualizations);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('CommenterVisualizations', function () {
	it('renders correctly', function () {

		var tree = /* istanbul ignore next */_reactTestRenderer2.default.
		create(
		/* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_CommenterVisualizations2.default, /* istanbul ignore next */{
			commenter: {
				nCommentsIliad: 0,
				nCommentsOdyssey: 0,
				nCommentsHymns: 0 },

			isTest: /* istanbul ignore next */true })).


		toJSON();
		expect(tree).toMatchSnapshot();
	});
}); // component: