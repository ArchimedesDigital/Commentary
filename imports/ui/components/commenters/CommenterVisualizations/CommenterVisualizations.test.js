import React from 'react';
import renderer from 'react-test-renderer';

// component:
import CommenterVisualizations from './CommenterVisualizations';

describe('CommenterVisualizations', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<CommenterVisualizations
					commenter={{
						nCommentsIliad: 0,
						nCommentsOdyssey: 0,
						nCommentsHymns: 0,
					}}
					isTest
				/>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
