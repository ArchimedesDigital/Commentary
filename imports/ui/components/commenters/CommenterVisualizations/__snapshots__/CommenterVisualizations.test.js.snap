// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommenterVisualizations renders correctly 1`] = `
<div
  className="commenter-visualizations"
>
  <div
    className="commenter-visualization-title"
  >
    <h2>
      Comments
    </h2>
  </div>
  <div
    className="commenter-work-circles"
  >
    
    
    
  </div>
  <div
    className="work-visualizations"
  >
    
  </div>
</div>
`;
